File: day1.py

def main():
    input = [int(line.strip()) for line in open("input/day1.txt")]
    count = 0
    for a, b in zip(input[:-1], input[1:]):
        if b > a:
            count += 1
    print(f"Part 1: {count}")

    # Part 2
    count2 = 0
    for i in range(len(input) - 2):
        if sum(input[i+1:i+4]) > sum(input[i:i+3]):
            count2 += 1
    print(f"Part 2: {count2}")


if __name__ == '__main__':
    main()

File: year2019/AdventOfCode/day01/Day1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace AdventOfCode.day01
{
    class Day1
    {
        public static void Solve()
        {
            var input = File.ReadAllLines("day01/day1.txt").ToList();
            var numInput = input.Select(double.Parse);
            double sum = 0;
            double sum2 = 0;
            foreach (var line in numInput)
            {
                sum += Math.Floor(line / 3) - 2;
            }
            Console.WriteLine("Part 1: {0}", sum);

            foreach (var line in numInput)
            {
                sum2 += CalculateFuel(line);
            }
            Console.WriteLine("Part 2: {0}", sum2);
        }
        static double CalculateFuel(double weight, double total = 0)
        {
            var result = Math.Floor(weight / 3) - 2;
            if (result > 0)
            {
                return CalculateFuel(result, total + result);
            }
            return total;
        }
    }
}
